chain/chain/src/types.rs

```diff
@@ -113,6 +113,8 @@ pub struct BlockHeaderInfo {
     pub chunk_mask: Vec<bool>,
     pub total_supply: Balance,
     pub latest_protocol_version: ProtocolVersion,
+    #[cfg(feature = "protocol_feature_rectify_inflation")]
+    pub timestamp: u64,
```

Comment:
Can we have some dedicated type for timestamps or at least indicate the timestamp granularity in the field name? It is really hard to reason about the timestamp whether it is seconds, microseconds, or what not.